lwcomponent/component.go

```diff
@@ -0,0 +1,212 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// A Lacework component package to help facilitate the loading and execution of components
+package lwcomponent
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+)
+
+type State struct {
+	Version    string      `json:"version"`
+	Components []Component `json:"components"`
+}
+
+func cacheDir() (string, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+
+	return path.Join(home, ".config", "lacework"), nil
+}
+
+// fileExists checks if a file exists and is not a directory
+func fileExists(filename string) bool {
```

Comment:
@afiune is it worth creating a library for this?

Reply:
I would be ok with that as long as we make it a private package (storing it at https://github.com/lacework/go-sdk/tree/main/internal)

Reply:
Created https://lacework.atlassian.net/browse/ALLY-844 for the introduction of private lwfile package.